shell/platform/android/io/flutter/embedding/android/FlutterImageView.java

```diff
@@ -40,13 +45,17 @@
  * onDraw}.
  */
 @TargetApi(19)
-public class FlutterImageView extends View implements RenderSurface {
+public class FlutterImageView extends View
+    implements RenderSurface, ImageReader.OnImageAvailableListener {
   private static final String TAG = "FlutterImageView";
 
   @NonNull private ImageReader imageReader;
-  @Nullable private Image currentImage;
   @Nullable private Bitmap currentBitmap;
   @Nullable private FlutterRenderer flutterRenderer;
+  @NonNull private final Set<Runnable> onImageAvailableListeners = new HashSet<>();
+  @NonNull private final List<Image> acquiredImages = new ArrayList<>();
```

Comment:
It's not clear to me why we need to keep the entire chain of images not closed? What problem is this solving?

Reply:
There is only 1 problem, if I don't do that, sometimes the image (HardwareBuffer) showing on the screen is not the image used to draw.
I tried to close the other image and only kept the current image open, it still happens.

Reply:
I re-checked and the problem is still not fixed this way. So I added code to close the unused image in `acquireLatestImage()`.

Reply:
I'm concerned we don't really understand what is happening here. More understanding is needed.

Reply:
My hypothesis is after drawing the Image and closing it at the next FrameCallback, the HardwareBuffer is freed and flutter renders the next frame on it, then it is composed to display on the screen. It matches the behavior. If anyone can confirm it.
I tested by increasing the max image to 4 and delaying the Image.close() 1 more FrameCallback, it does fix the problem.
But it occurs rarely and is very hard to notice so I think no need to change the code.

Reply:
I replaced this with a new implementation which makes more sense. The image is closed once the associated Bitmap is no longer in use. The max image is changed to 4 because I got error calling `acquireLatestImage()` when 2 images are open.